release/releaser.go

```diff
@@ -154,6 +154,16 @@ func (r *Releaser) release(instanceID flux.InstanceID, job *jobs.Job, logStatus
 	applyErr := applyChanges(rc.Instance, updates, results)
 	timer.ObserveDuration()
 
+	// #492: If updating all services, but requested a specific image, only result in services that pertain to that image
+	if spec.ServiceSpec == flux.ServiceSpecAll && spec.ImageSpec != flux.ImageSpecNone {
+		for k, v := range results {
+			if v.Status == flux.ReleaseStatusSkipped || v.Status == flux.ReleaseStatusIgnored {
+				delete(results, k)
+			}
+		}
+
+	}
```

Comment:
afaict, this is saying
```Go
if --all && --update-image=xxx {
  hide all skipped/ignored services
}
```
Which is not quite right. Skipped/ignored services should still be included, but only if they would have been modified (i.e. if they use the image in question).